sos/report/plugins/shmcli.py

```diff
@@ -0,0 +1,105 @@
+
+
+class SHMcli(Plugin, IndependentPlugin):
+    """shmcli pulls hardware information from PowerVault/Dell
+    Storage JBOD's attached to server.
+    It provides information of the adapters, emms, drives,
+    enclosures, fans, power supplies and the sensory data of
+    temp, voltage, and current sensors.
+    """
+
+    short_desc = 'Dell Server Hardware Manager'
+
+    plugin_name = 'shmcli'
+    profiles = ('system', 'storage', 'hardware',)
+
+    files = ('/opt/dell/ServerHardwareManager/' +
+             'ServerHardwareManagerCLI/bin/shmcli',)
+
+    option_list = [
+        PluginOpt('debug', default=False, desc='capture support debug data')
+    ]
+
+    def setup(self):
+        cmd = "/opt/dell/ServerHardwareManager/" \
+              "ServerHardwareManagerCLI/bin/shmcli"
+
+        subcmds = [
+            'list adapters',
+            'list physical enclosures',
+            'list failed drives'
+        ]
+
+        for subcmd in subcmds:
+            self.add_cmd_output(
+                "%s %s" % (cmd, subcmd),
+                suggest_filename="shmcli_%s" % (subcmd))
+
+        models = []
+
+        # Get the storage hardware models
+        result = self.exec_cmd('lsscsi -g')
+        if result['status'] == 0:
+            for line in result['output'].splitlines():
+                if (line.split()[2] == 'DELL'):
+                    models.append(line.split()[3])
+        models = list(set(models))
+
+        subcmds = [
+            'list emms',
+            'list drawers',
+            'list emm slots',
+            'list drive slots',
+            'list fans',
+            'list temp sensors',
+            'list voltage sensors',
+            'list current sensors',
+            'list power supplies',
+            'info enclosure'
+        ]
+
+        result = self.collect_cmd_output(
+            '%s list enclosures' % (cmd),
+            suggest_filename='shmcli_list_enclosures'
+        )
+        if result['status'] == 0:
+            for line in result['output'].splitlines()[2:-2]:
+                _line = line.split()
+                if any(m in _line for m in models):
+                    adapt_index = _line[-1]
+                    enc_index = _line[0]
+                    for subcmd in subcmds:
+                        _cmd = ("%s %s -a=%s -enc=%s"
+                                % (cmd, subcmd, adapt_index, enc_index))
+                        _fname = _cmd.replace(cmd, 'shmcli')
+                        self.add_cmd_output(_cmd, suggest_filename=_fname)
+                    if self.get_option('debug'):
+                        logpath = self.get_cmd_output_path(make=False)
+                        _dcmd = ("%s getdebugcli -a=%s -enc=%s -outputdir=%s"
+                                 % (cmd, adapt_index, enc_index, logpath))
+                        _dname = _dcmd.replace(cmd, 'shmcli')
+                        self.add_cmd_output(_dcmd, suggest_filename=_dname)
```

Comment:
Looking at this again.

First, a question - what does this output look like? What's getting dropped into the plugin dir by the `getdebugcli` command?

Second, seeing the screenshot again let's not have the outputdir as part of the filename. Especially since it wouldn't make sense in the context of the plugin dir. We're probably good to just use (effectively) `suggest_filename=shmcli getbugcli -a=%s -enc=%s`

Reply:
Regarding the first part,
After the executing  `getdebugcli` cmd a single file with `.out` format would be available,
<img width="679" alt="Screenshot 2022-05-13 at 9 44 39 AM" src="https://user-images.githubusercontent.com/66179696/168226309-b83ad8fd-ea52-4ff9-8075-876b2baa55f6.png">

And,
I have updated the approach.